ext/lvgl/lvgl.lb

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020, Raphael Lehmann
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+def init(module):
+    module.name = ":lvgl"
+    module.description = """
+# LVGL graphics library
+
+LVGL is a free and open-source graphics library providing everything you need
+to create embedded GUI with easy-to-use graphical elements, beautiful visual effects
+and low memory footprint.
+
+- https://lvgl.io/
+- https://github.com/lvgl/lvgl
+"""
+
+def prepare(module, options):
+
+    module.add_option(
+        NumericOption(
+            name="resolution_horizontal_max",
+            description="Maximal horizontal resolution to support by lvgl",
+            minimum=1,
+            default=480))
+    module.add_option(
+        NumericOption(
+            name="resolution_vertical_max",
+            description="Maximal vertical resolution to support by lvgl",
+            minimum=1,
+            default=320))
+    module.add_option(
+        EnumerationOption(
+            name="color_depth",
+            description="Color depth",
+            enumeration={
+                "OneBitPerPixel": 1,
+                "RGB233": 8,
+                "RGB565": 16,
+                "ARGB8888": 32,
+                },
+            default="RGB565"))
+    module.add_option(
+        NumericOption(
+            name="dpi",
+            description="Dot Per Inch: used to initialize default sizes",
+            minimum=1,
+            default=100))
+    module.add_option(
+        EnumerationOption(
+            name="logging",
+            description="Verbosity of the log",
+            enumeration={
+                "Debug": "LV_LOG_LEVEL_TRACE",
+                "Info": "LV_LOG_LEVEL_INFO",
+                "Warning": "LV_LOG_LEVEL_WARN",
+                "Error": "LV_LOG_LEVEL_ERROR",
+                "Disabled": "LV_LOG_LEVEL_NONE",
+                },
+            default="Disabled"))
+
+    if options[":target"].identifier.platform != "hosted":
+        module.depends(":platform:heap")
```

Comment:
Otherwise building hosted with the `:lvgl` module enabled (like the doxypress docs generator does) will fail because the `:platform:heap` module does not exist for hosted.